src/components/input/index.tsx

```diff
@@ -84,10 +86,21 @@ export class SmoothlyInput implements Clearable, Input, Editable {
 		if (value != before) {
 			if (typeof value == "string")
 				value = value.trim()
-			this.smoothlyInput.emit({ [this.name]: value })
+			this.smoothlyInputEmit(value)
 		}
 		this.listener.changed?.(this)
 	}
+	async smoothlyInputEmit(value: any): Promise<void> {
+		this.abortInputEvent?.()
+		const t: ReturnType<typeof setTimeout> = setTimeout(() => {
```

Comment:
if you use `window.setTimeout()` instead of just `timeout()` you will get the correct return type of `number`. This is due to Node using non standard implementation of `setTimeout` and Typescript / VSC doesnt know which environment our javascript runs in

Reply:
i also suspec that this is not suppsoed to be called t?